packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -165,77 +170,42 @@ export enum SObjectCategory {
 
 type SubRequest = { method: string; url: string };
 type BatchRequest = { batchRequests: SubRequest[] };
-
 type SubResponse = { statusCode: number; result: SObject };
-
 type BatchResponse = { hasErrors: boolean; results: SubResponse[] };
 
+export class ForceListSObjectSchemaExecutor {
+  public build(type: string): Command {
+    return new SfdxCommandBuilder()
+      .withArg('force:schema:sobject:list')
+      .withFlag('--sobjecttypecategory', type)
+      .withJson()
+      .build();
+  }
+
+  public execute(projectPath: string, type: string): CliCommandExecution {
+    const execution = new CliCommandExecutor(this.build(type), {
+      cwd: projectPath
+    }).execute();
+    return execution;
+  }
+}
+
 export class SObjectDescribe {
   private accessToken?: string;
   private instanceUrl?: string;
   private readonly servicesPath: string = 'services/data';
   // the targetVersion should be consistent with the Cli even if only using REST calls
-  private targetVersion = '';
+  private readonly targetVersion = '46.0';
   private readonly versionPrefix = 'v';
   private readonly sobjectsPart: string = 'sobjects';
   private readonly batchPart: string = 'composite/batch';
 
-  public async describeSObject(
-    projectPath: string,
-    type: string,
-    username?: string
-  ): Promise<SObject> {
-    await this.setupConnection(projectPath, username);
-
-    const urlElements = [
-      this.instanceUrl,
-      this.servicesPath,
-      this.getVersion(),
-      this.sobjectsPart,
-      type,
-      'describe'
-    ];
-
-    const requestUrl = urlElements.join('/');
-
-    const options: XHROptions = {
-      type: 'GET',
-      url: requestUrl,
-      headers: {
-        'Content-Type': 'application/json',
-        Accept: 'application/json',
-        Authorization: `OAuth ${this.accessToken}`,
-        'User-Agent': 'salesforcedx-extension',
-        'Sforce-Call-Options': `client=${CLIENT_ID}`
-      }
-    };
-
-    try {
-      const response: XHRResponse = await xhr(options);
-      const sobject = JSON.parse(response.responseText) as SObject;
-      return Promise.resolve(sobject);
-    } catch (error) {
-      const xhrResponse: XHRResponse = error;
-      return Promise.reject(xhrResponse.responseText);
-    }
-  }
-
   public async describeGlobal(
     projectPath: string,
-    type: SObjectCategory,
-    username?: string
+    type: SObjectCategory
   ): Promise<string[]> {
-    const builder = new SfdxCommandBuilder()
-      .withArg('force:schema:sobject:list')
-      .withFlag('--sobjecttypecategory', type.toString());
-    if (username) {
```

Comment:
Yes, that's because I noticed that the username parameter was never being sent by the code calling the describe command and decided to update the interfaces in order to remove the `username` parameter.